assets/js/modules/analytics-4/utils/withCustomDimensions.test.js

```diff
@@ -89,6 +98,53 @@ describe( 'withCustomDimensions', () => {
 		expect( container ).toHaveTextContent( 'Insufficient permissions' );
 	} );
 
+	it( 'sets appropriate `redirectURL` in persmission error object if creating custom dimensions failed due to the user not having `EDIT_SCOPE`', () => {
+		registry.dispatch( MODULES_ANALYTICS_4 ).setSettings( {
+			propertyID: '123456789',
+			availableCustomDimensions: [ customDimension ],
+		} );
+
+		provideUserAuthentication( registry, {
+			grantedScopes: [],
+		} );
+
+		const error = {
+			code: 'test-error-code',
+			message: 'Test error message',
+			data: {
+				reason: ERROR_REASON_INSUFFICIENT_PERMISSIONS,
+			},
+		};
+
+		provideCustomDimensionError( registry, {
+			customDimension,
+			error,
+		} );
+
+		const { queryByText, getByRole } = render(
+			<WithCustomDimensionsComponent />,
+			{
+				registry,
+			}
+		);
+
+		expect( queryByText( /retry/i ) ).toBeInTheDocument();
```

Comment:
As above:

```suggestion
		expect( getByText( /retry/i ) ).toBeInTheDocument();
```

Reply:
@techanvil I left `queryByText` in second test case, as text is not present, hence `getByText` throws error

Reply:
Hmm, possibly a case of crossed wired here - we are checking the element _is_ in the document in this case, so `getByText()` should work (and does in my testing). I'll apply the change myself to save the round trip.